### PR TITLE
DRIVERS-2789 Use Markdown for Specifications Documentation

### PR DESCRIPTION
**Pull request by blink1073**
Please complete the following before merging:

- [x] Update changelog.
- [ ] Make sure there are generated JSON files from the YAML test files.
- [ ] Test changes in at least one language driver.
- [ ] Test these changes against all server versions and topologies (including standalone, replica set, sharded clusters, and serverless).

### CODE REVIEW

File: source/client-side-operations-timeout/client-side-operations-timeout.md [hunk 1]
```diff
@@ -0,0 +1,906 @@
...
+
+______________________________________________________________________
+
+## Abstract
+
+This specification outlines a new `timeoutMS` option to govern the
+amount of time that a single operation can execute before control is
+returned to the user. This timeout applies to all of the work done to
+execute the operation, including but not limited to server selection,
+connection checkout, and server-side execution.
```
Comment (jmikola) [inline source/client-side-operations-timeout/client-side-operations-timeout.md:14]: Did you plan on enforcing the line length with linting tools? Knowing how fast and loose folks are when editing specs (I've seen entire paragraphs written without line breaks), I think it'd be a matter of time before the line lengths get inconsistent.

Also, any thoughts on 120 vs. 80? When link syntax is involved, the 80-column limit results in some very aggressive wrapping.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+## Prose test numbering
+
+When numbering prose tests, always use relative numbered bullets (`#.`).
+New tests must be appended at the end of the test list, since drivers
+may refer to existing tests by number.
+
+Outdated tests must not be removed completely, but may be marked as such
+(e.g. by striking through or replacing the entire test with a note (e.g.
+**Removed**).
```
Comment (jmikola) [inline README.md:130]: This RST sytnax should be revised. Equivalent in Markdown would be `*Removed*` I think.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+To run a manual hook like `rstcheck` manually, run:
+
+```bash
+pre-commit run --all-files --hook-stage manual rstcheck
+```
+
+## Prose test numbering
+
+When numbering prose tests, always use relative numbered bullets (`#.`).
```
Comment (jmikola) [inline README.md:124]:
```suggestion
When numbering prose tests, always use relative numbered bullets (`1.`).
```

Switching to Markdown.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+To set up `pre-commit` locally, run:
+
+```bash
+brew install pre-commit
+pre-commit install
+```
+
+To run `pre-commit` manually, run `pre-commit run --all-files`.
+
+To run a manual hook like `rstcheck` manually, run:
```
Comment (jmikola) [inline README.md:116]: Is this still relevant? I think it'd be forgotten if left until the last spec is migrated to Markdown.

File: source/client-side-operations-timeout/client-side-operations-timeout.rst [hunk 5]
```diff
@@ -1,935 +0,0 @@
-==============================
-Client Side Operations Timeout
-==============================
-
-:Status: Accepted
-:Minimum Server Version: 2.6
-
-.. contents::
```
[the remark this answers is not in the document]
Comment (blink1073) [reply source/client-side-operations-timeout/client-side-operations-timeout.rst:8]: Yes:

image

File: README.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+Backward breaking behavior changes can be more dangerous and disruptive
+than backward breaking API changes. When thinking about the implications
+of a behavior change, ask yourself what could happen if a user upgraded
+your library without carefully reading the changelog and/or adequately
+testing the change.
+
+## Writing Documents
+
+Write documents using
+[reStructuredText](http://docutils.sourceforge.net/rst.html), following
```
[the remark this answers is not in the document]
Comment (blink1073) [reply README.md:95]: Done

File: source/uri-options/uri-options.rst [hunk 7]
```diff
@@ -315,7 +315,7 @@ pertaining to URI options apply here.
      - non-negative integer; 0 means no timeout
      - no timeout
      - no
-     - NOTE: This option is deprecated  in favor of `timeoutMS <../client-side-operations-timeout/client-side-operations-timeout.rst#timeoutMS>`_
+     - NOTE: This option is deprecated  in favor of `timeoutMS <../client-side-operations-timeout/client-side-operations-timeout.md#timeoutMS>`_
```
[the remark this answers is not in the document]
Comment (blink1073) [reply source/uri-options/uri-options.rst:318]: We will be enforcing links once everything is converted to Markdown, but based on my testing in my gist, anchor hrefs are case-insensitive.

File: source/client-side-operations-timeout/client-side-operations-timeout.md (hunk 1 above)
Comment (blink1073) [reply source/client-side-operations-timeout/client-side-operations-timeout.md:14]: Line length is not enforceable with `mdformat`.

File: README.md (hunk 2 above)
Comment (blink1073) [reply README.md:130]: Done

File: README.md (hunk 4 above)
Comment (blink1073) [reply README.md:116]: Updated

File: source/change-streams/change-streams.rst [hunk 8]
```diff
@@ -836,7 +836,7 @@ All drivers MUST document how users can iterate a change stream and receive *all
 Timeouts
 ^^^^^^^^
 
-Drivers MUST apply timeouts to change stream establishment, iteration, and resume attempts per `Client Side Operations Timeout: Change Streams <../client-side-operations-timeout/client-side-operations-timeout.rst#Change-Streams>`__.
+Drivers MUST apply timeouts to change stream establishment, iteration, and resume attempts per `Client Side Operations Timeout: Change Streams <../client-side-operations-timeout/client-side-operations-timeout.md#Change-Streams>`__.
```
Comment (jmikola) [inline source/change-streams/change-streams.rst:839]:
```suggestion
Drivers MUST apply timeouts to change stream establishment, iteration, and resume attempts per `Client Side Operations Timeout: Change Streams <../client-side-operations-timeout/client-side-operations-timeout.md#change-streams>`__.
```

I think all of the generated headings are lowercase, unless GitHub creates multiple variations for each heading. I'm checking this by viewing the rendered copy of the ported CSOT spec (e.g. [client-side-operations-timeout.md](https://github.com/mongodb/specifications/blob/1d074a35ab9bcd4496aae82ced5ae0b3a0ebd62f/source/client-side-operations-timeout/client-side-operations-timeout.md)).

File: source/uri-options/uri-options.rst [hunk 9]
```diff
@@ -343,7 +343,7 @@ pertaining to URI options apply here.
    *
      - timeoutMS
      - non-negative integer; 0 or unset means no timeout
-     - Defined in `Client Side Operations Timeout: timeoutMS <../client-side-operations-timeout/client-side-operations-timeout.rst#timeoutMS>`_.
+     - Defined in `Client Side Operations Timeout: timeoutMS <../client-side-operations-timeout/client-side-operations-timeout.md#timeoutMS>`_.
```
Comment (jmikola) [inline source/uri-options/uri-options.rst:346]:
```suggestion
     - Defined in `Client Side Operations Timeout: timeoutMS <../client-side-operations-timeout/client-side-operations-timeout.md#timeoutms>`_.
```

There are a few other instances of this permalink that were missed.

File: source/uri-options/uri-options.rst [hunk 10]
```diff
@@ -422,7 +422,7 @@ pertaining to URI options apply here.
      - positive number
      - defined in the `Connection Pooling spec`_
      - required for drivers with connection pools, with exceptions described in the `Connection Pooling spec`_
-     - NOTE: This option is deprecated  in favor of `timeoutMS <../client-side-operations-timeout/client-side-operations-timeout.rst#timeoutMS>`_
+     - NOTE: This option is deprecated  in favor of `timeoutMS <../client-side-operations-timeout/client-side-operations-timeout.md#timeoutMS>`_
```
Comment (jmikola) [inline source/uri-options/uri-options.rst:425]:
```suggestion
     - NOTE: This option is deprecated  in favor of `timeoutMS <../client-side-operations-timeout/client-side-operations-timeout.md#timeoutms>`_
```

File: source/uri-options/uri-options.rst [hunk 11]
```diff
@@ -431,7 +431,7 @@ pertaining to URI options apply here.
      - non-negative 64-bit integer; 0 means no timeout
      - no timeout
      - no
-     - NOTE: This option is deprecated  in favor of `timeoutMS <../client-side-operations-timeout/client-side-operations-timeout.rst#timeoutMS>`_
+     - NOTE: This option is deprecated  in favor of `timeoutMS <../client-side-operations-timeout/client-side-operations-timeout.md#timeoutMS>`_
```
Comment (jmikola) [inline source/uri-options/uri-options.rst:434]:
```suggestion
     - NOTE: This option is deprecated  in favor of `timeoutMS <../client-side-operations-timeout/client-side-operations-timeout.md#timeoutms>`_
```

**Review by jmikola — APPROVED** [review]
A few more comments about link anchors with the wrong character case.

And I'm still in favor of 120 instead of 80 for character line limits on spec docs; the last comment in https://github.com/mongodb/specifications/pull/1482#discussion_r1423932026 didn't acknowledge my question about that.

Happy to defer review to the actual CSOT spec owner(s) to sort the rest of this out.

**Comment by blink1073** [comment]
> And I'm still in favor of 120 instead of 80 for character line limits on spec docs

Fixed.

**Review by ShaneHarvey — APPROVED** [review]
LGTM

**Pull request merged**